src/dialogs/enter-code/dialog-enter-code.ts

```diff
@@ -52,8 +52,14 @@ export class DialogEnterCode
   }
 
   private _submit(): void {
-    this._dialogParams?.submit?.(this._input?.value ?? "");
-    this.closeDialog();
+    if (
+      this._dialogParams &&
+      this._dialogParams.codePattern &&
+      RegExp(this._dialogParams.codePattern).test(this._input?.value ?? "")
```

Comment:
If there is code pattern the user can not submit? Is code pattern mandatory in core? Looking at the dev docs it seems to be optional?